### ISSUE TITLE
[YAML] Everything is "String" colored now

### ISSUE THREAD

**Issue by keldin-coding** [sublimehq/Packages#3288]
### What happened?

As of updating to the latest sublimeText, it seems like even `keys` of Hashes/Dictionaries are now highlighted as if they were regular strings. Before, they used to be differentiated (pink vs green in my specific color scheme) making it easier to identify key/value pairings. I would guess this is related to #3272, but I can't be sure as I've never dug through the syntax highlighting code before.

<img width="302" alt="Screen Shot 2022-03-23 at 13 41 44" src="https://user-images.githubusercontent.com/8674577/159791860-4c079ae2-31b0-42cd-8057-93d0189ab5dd.png">

**Comment by deathaxe** [comment]
This scope naming change has been made by intent to get inline with how JSON keys are scoped for years now.

**Comment by lirossarvet** [comment]
And coming back to this now, even the built-in Monokai didn't get updated. Well enough, I think I can customise the UI, was just surprising all of a sudden

**Comment by deathaxe** [comment]
See https://github.com/sublimehq/sublime_text/issues/3590 for built-in color schemes, which has lasted for years as well.

The point is, what you see in YAML is "state of the art" in JSON for years.

**Comment by deathaxe** [comment]
There's also the following statement about it: https://discord.com/channels/280102180189634562/280102180189634562/956253218034315304

**Comment by lirossarvet** [comment]
Fair enough. I'll leave the same fix here as is in the issue I linked and is in the discord which worked for Monokai:

```json
{
            "name": "Mapping Key Names",
            "scope": "meta.mapping.key string",
            "foreground": "var(red2)"
        }

```

**Comment by lirossarvet** [comment]
Thanks for the help and confirmation! :)

**Comment by rgpublic** [comment]
For anyone wondering where to apply the fix mentioned by @lirossarvet:

Just call Preferences => Customize Color Scheme

and add this between the square brackets under "rules":

```
        {
            "name": "Mapping Key Names",
            "scope": "meta.mapping.key string",
            "foreground": "#ff79c6"
        }
```

This will color everything like it used to be. Really weird change, by the way. Clearly a regression if you ask me. The syntax highlighting is almost useless without this. I thought it was broken before I found this issue.

**Comment by levonet** [comment]
I copied the `YAML.sublime-package` from the previous version. This temporarily solved the problem. But I would like this problem to be solved with the next version of ST.

**Comment by deathaxe** [comment]
It's your color scheme, which needs to be tweaked. Syntax definition won't be reverted.

YAML applies the same scopes to keys as JSON has done for ages. Everyone how complains about how YAML looks should have noticed it for JSON long before.

**Comment by levonet** [comment]
@deathaxe 
> YAML applies the same scopes to keys as JSON has done for ages. Everyone how complains about how YAML looks should have noticed it for JSON long before.

There is a significant difference between JSON and YAML. The first format was created with love for machines and hate for people. The second is the opposite.

**Comment by h0jeZvgoxFepBQ2C** [comment]
Yeah this really looks like a bug, came here to find a solution for the broken syntax highlighting.

**Comment by h0jeZvgoxFepBQ2C** [comment]
<img width="871" alt="Bildschirmfoto 2022-11-11 um 10 26 09" src="https://user-images.githubusercontent.com/10075/201309383-4c93f60f-613f-4562-bf9d-f3b169e57a28.png">


Could someone tell me where to add this fix, I don't have a "Customize Colorscheme" option?

**Comment by jfcherng** [comment]
> Also this broken appearance just happened yesterday for me with update to the new Sublime release, not before?

It first happens on Mar 24, 2022 as you can see in this thread. However, that's a dev build. If you use stable build, it happens yesterday.

**Comment by h0jeZvgoxFepBQ2C** [comment]
Ah I see thanks, I'm using following for the railscasts theme:

```json
    {
      "name": "Mapping Key Names",
      "scope": "meta.mapping.key string",
      "foreground": "#FFC66D"
    }
```

**Comment by pasqualguerrero** [comment]
> Ah I see thanks, I'm using following for the railscasts theme:
> 
> ```json
>     {
>       "name": "Mapping Key Names",
>       "scope": "meta.mapping.key string",
>       "foreground": "#FFC66D"
>     }
> ```

If you want you can directly edit your `*.tmTheme` to not have config in separated files:
```xml
<dict>
    <key>name</key>
    <string>Mapping Key Names</string>
    <key>scope</key>
    <string>meta.mapping.key string</string>
    <key>settings</key>
    <dict>
        <key>foreground</key>
        <string>#FFC66D</string>
    </dict>
</dict>
```

**Comment by geerlingguy** [comment]
It's a bit confusing since all the built-in color schemes seem to have 'broken' (even if they're not 'broken', YAML is incredibly hard to parse now, where it was not before, at least using Cobalt and testing on a couple other built-in color schemes).

**Comment by deathaxe** [comment]
Updates requests for default color schemes are still pending (https://github.com/sublimehq/sublime_text/issues/3590).

**Comment by michaelblyons** [comment]
Any remaining watchers: https://github.com/sublimehq/sublime_text/issues/3590 is now completed in a Dev build.
